File: EcoJardimPro - BackEnd/EcoJardimProject/EcoJardimProject.Domain/Entities/Projeto.cs
using EcoJardimProject.Domain.Base;

namespace EcoJardimProject.Domain.Entities;

public class Projeto : IEntity
{
    public long Id { get; set; }
    public string Nome { get; set; }
    public string Descricao { get; set; }
    public long StatusId { get; set; }
    public Status? Status { get; private set; }
    public DateTime PrazoInicial { get; set; }
    public DateTime PrazoFinal { get; set; }
    public decimal Valor { get; set; }
    public long ClienteId { get; set; }
    public Cliente Cliente { get; set; }
    
    
    public List<string> ValidateProjeto()
    {
        var errors = new List<string>();

        if (string.IsNullOrEmpty(Nome))
        {
            errors.Add("Nome do Projeto é obrigatório");
        }

        if (string.IsNullOrEmpty(Descricao))
        {
            errors.Add("A Descrição do projeto é obrigatória");
        }
        
        if (Valor <= 0)
        {
            errors.Add("O Valor do projeto é obrigatório");
        }
        return errors;
    }
}

File: EcoJardimPro - FronEnd/ecojardimprofrontend/src/models/projeto/Projeto.ts

import Cliente from "../cliente/Cliente";
import Status from "../status/Status";


export default class Projeto{
    public id: number;
    public nome: string;
    public descricao: string;
    public statusId!: number;
    public status: Status
    public prazoInicial:string;
    public prazoFinal: string;
    public valor: number;
    public clienteId: number;
    public cliente: Cliente;

    constructor() {
        this.id = 0
        this.nome = ''
        this.descricao = ''
        this.statusId = 0
        this.prazoInicial = '0001-01-01T00:00:00'
        this.prazoFinal = '0001-01-01T00:00:00'
        this.valor = 0
        this.clienteId = 0
        this.cliente = new Cliente()
    }
}